src/SprykerEco/Zed/AmazonPay/Business/Payment/Handler/Transaction/AuthorizeTransaction.php

```diff
@@ -63,6 +64,9 @@ protected function updateAuthorizationReferenceId(AmazonpayCallTransfer $amazonP
     protected function updatePaymentEntity(AmazonpayCallTransfer $amazonPayCallTransfer)
     {
         if (!$this->isPaymentSuccess($amazonPayCallTransfer)) {
+            $this->paymentEntity->setStatus(AmazonPayConfig::STATUS_DECLINED);
```

Comment:
entity on the business layer... and as property of the model...
But I guess this is a legacy code.